SampleFuzzingJobs/InPlaceFuzzing/inplacefuzzing/inplacefuzzing/inplacefuzzing.cpp

```diff
@@ -0,0 +1,110 @@
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
+
+// File header definition
+struct Header {
+	char* constant;
+	char* searchString;
+	int numberOfReferenceFiles;
+};
+
+// File format definition
+struct FileFormat {
+	Header header;
+	char** files;
+};
+
+
+// Load file from the specified path
+// Do not check for any errors
+FileFormat loadFile(char* filePath) {
+	FILE* pSeed = fopen(filePath, "rb");
+
+	fseek(pSeed, 0, SEEK_END);
+	long size = ftell(pSeed);
+	rewind(pSeed);
+
+	char* buffer = (char*)malloc(sizeof(char) * size + 1);
+	memset(buffer, NULL, sizeof(char) * (size + 1));
+	fread(buffer, sizeof(char), size, pSeed);
+
+	Header header = { NULL, NULL, -1 };
+
+	char* pch = strtok(buffer, "\n");
+	while (pch != NULL)
+	{
+		int len = strlen(pch);
+
+		if (header.constant == NULL) {
+			header.constant = (char*)malloc(sizeof(char) * (len + 1));
+			memset(header.constant, NULL, len + 1);
+			memcpy(header.constant, pch, len);
+		}
+		else if (header.searchString == NULL) {
+			header.searchString = (char*)malloc(sizeof(char) * (len + 1));
+			memset(header.searchString, NULL, len + 1);
+			memcpy(header.searchString, pch, len);
+		}
+		else {
+			header.numberOfReferenceFiles = atoi(pch);
+			break;
+		}
+		pch = strtok(NULL, "\n");
+	}
+	FileFormat fileFormat = { header, (char**)malloc(sizeof(char*) * header.numberOfReferenceFiles)};
+	for (int i = 0; i < header.numberOfReferenceFiles; i++) {
+		pch = strtok(NULL, "\n");
+		int len = strlen(pch);
+		fileFormat.files[i] = (char*)malloc(sizeof(char) * (len + 1));
```

Comment:
yeah, that's right! we don't even bother freeing this memory. That besides the point of this sample, and if we really cared about doing things properly we would probably not use C/C++ ;-)

Reply:
memory gets freed when the process exits, since the plan is to load all files and keep them till end of the process execution

Reply:
Yes, I know... though I was not expecting such argument from an F# developer.
Leaving it to process exit for OS to cleanup is just wrong from a program-compositionality point of view. (think about how code evolves and gets reused over time)

Reply:
ok, I'll add cleanup

Reply:
No need, Fine to keep like this. This sample is not meant to be production quality.